rules/aip0131/method_rules_test.go

```diff
@@ -92,3 +96,158 @@ func TestResponseMessageName(t *testing.T) {
 		})
 	}
 }
+
+func TestHttpVerb(t *testing.T) {
+	// Set up testing permutations.
+	tests := []struct {
+		testName   string
+		verb       string
+		methodName string
+		msg        string
+	}{
+		{"Valid", "get", "GetBook", ""},
+		{"Invalid", "post", "GetBook", "HTTP GET"},
+		{"Irrelevant", "post", "AcquireBook", ""},
+	}
+
+	// Run each test.
+	for _, test := range tests {
+		t.Run(test.testName, func(t *testing.T) {
+			// Create the appropriate pattern based on the desired verb.
+			var httpRule *annotations.HttpRule
+			if test.verb == "post" {
+				httpRule = &annotations.HttpRule{
+					Pattern: &annotations.HttpRule_Post{
+						Post: "/v1/{name=publishers/*/books/*}",
+					},
+				}
+			} else {
+				httpRule = &annotations.HttpRule{
+					Pattern: &annotations.HttpRule_Get{
+						Get: "/v1/{name=publishers/*/books/*}",
+					},
+				}
+			}
+
+			// Create a MethodOptions with the annotation set.
+			opts := &dpb.MethodOptions{}
+			if err := proto.SetExtension(opts, annotations.E_Http, httpRule); err != nil {
+				t.Fatalf("Failed to set google.api.http annotation.")
+			}
+
+			// Create a minimal service with a AIP-131 Get method
+			// (or with a different method, in the "Irrelevant" case).
+			service, err := builder.NewService("Library").AddMethod(builder.NewMethod(test.methodName,
+				builder.RpcTypeMessage(builder.NewMessage("GetBookRequest"), false),
+				builder.RpcTypeMessage(builder.NewMessage("Book"), false),
+			).SetOptions(opts)).Build()
+			if err != nil {
+				t.Fatalf("Could not build %s method.", test.methodName)
+			}
+
+			// Run the method, ensure we get what we expect.
+			problems := httpVerb.Lint(service.GetFile())
+			if test.msg == "" && len(problems) > 0 {
+				t.Errorf("Got %v, expected no problems.", problems)
+			} else if test.msg != "" && !strings.Contains(problems[0].Message, test.msg) {
```

Comment:
What about `problems` is empty?

Reply:
Then it would panic. I am fine with that.

In general, still not a fan of the test situation. I plan to do more work to make it easier to test problems.